sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala

```diff
@@ -265,4 +267,30 @@ class IntervalUtilsSuite extends SparkFunSuite {
         assert(e.getMessage.contains("divide by zero"))
     }
   }
+
+  test("from bounded day-time string") {
+    val input = "5 12:40:30.999999999"
+
+    def check(from: IntervalUnit, to: IntervalUnit, expected: String): Unit = {
+      withClue(s"from = $from, to = $to") {
+        assert(fromDayTimeString(input, from, to) === fromString(expected))
+      }
+    }
+
+    withSQLConf(SQLConf.DIALECT.key -> "Spark") {
```

Comment:
can we have a test case to show the error behavior?

Reply:
Just to be clear, this PR just truncates results according to specified bound. In particular, it takes into account the left bound. Could you explain, please, what do you mean by saying `error behavior`?

Reply:
I think the conclusion is to fail for invalid bounds?

Reply:
Do you mean if a string doesn't match to specified bounds like in `literals.sql`: https://github.com/apache/spark/pull/26358/files#diff-4f9e28af8e9fcb40a8a99b4e49f3b9b2R424 ?

Reply:
yea, like what Oracle does.

Reply:
Here is the PR with strict parsing https://github.com/apache/spark/pull/26473